### ISSUE TITLE
Build is broken on Debian 11

### ISSUE THREAD

**Issue by dpvpro** [c-ares/c-ares#416]
The same error like in #409

Build output:

```
user@debian-build:/veil_rebuild/test$ apt source -b c-ares
Reading package lists... Done
NOTICE: 'c-ares' packaging is maintained in the 'Git' version control system at:
https://salsa.debian.org/debian/c-ares.git
Please use:
git clone https://salsa.debian.org/debian/c-ares.git
to retrieve the latest (possibly unreleased) updates to the package.
Need to get 1,531 kB of source archives.
Get:1 http://security.debian.org/debian-security bullseye-security/main c-ares 1.17.1-1+deb11u1 (dsc) [2,301 B]
Get:2 http://security.debian.org/debian-security bullseye-security/main c-ares 1.17.1-1+deb11u1 (tar) [1,519 kB]
Get:3 http://security.debian.org/debian-security bullseye-security/main c-ares 1.17.1-1+deb11u1 (asc) [488 B]
Get:4 http://security.debian.org/debian-security bullseye-security/main c-ares 1.17.1-1+deb11u1 (diff) [9,960 B]
Fetched 1,531 kB in 0s (3,756 kB/s)
dpkg-source: info: extracting c-ares in c-ares-1.17.1
dpkg-source: info: unpacking c-ares_1.17.1.orig.tar.gz
dpkg-source: info: unpacking c-ares_1.17.1-1+deb11u1.debian.tar.xz
dpkg-source: info: using patch list from debian/patches/series
dpkg-source: info: applying disable-cflags-rewrite.diff
dpkg-source: info: applying ares_expand_name-should-escape-more-characters.patch
dpkg-source: info: applying ares_expand_name-fix-formatting-and-handling-of-root.patch
dpkg-buildpackage: info: source package c-ares
dpkg-buildpackage: info: source version 1.17.1-1+deb11u1
dpkg-buildpackage: info: source distribution bullseye-security
dpkg-buildpackage: info: source changed by Salvatore Bonaccorso <carnil@debian.org>
dpkg-buildpackage: info: host architecture amd64
 dpkg-source --before-build .
 fakeroot debian/rules clean
dh clean
   dh_clean
 debian/rules build
dh build
   dh_update_autotools_config
   dh_autoreconf
aclocal: installing 'm4/ax_ac_append_to_file.m4' from '/usr/share/aclocal/ax_ac_append_to_file.m4'
aclocal: installing 'm4/ax_ac_print_to_file.m4' from '/usr/share/aclocal/ax_ac_print_to_file.m4'
aclocal: installing 'm4/ax_add_am_macro_static.m4' from '/usr/share/aclocal/ax_add_am_macro_static.m4'
aclocal: installing 'm4/ax_am_macros_static.m4' from '/usr/share/aclocal/ax_am_macros_static.m4'
aclocal: installing 'm4/ax_check_gnu_make.m4' from '/usr/share/aclocal/ax_check_gnu_make.m4'
aclocal: overwriting 'm4/ax_code_coverage.m4' with '/usr/share/aclocal/ax_code_coverage.m4'
aclocal: installing 'm4/ax_cxx_compile_stdcxx.m4' from '/usr/share/aclocal/ax_cxx_compile_stdcxx.m4'
aclocal: overwriting 'm4/ax_cxx_compile_stdcxx_11.m4' with '/usr/share/aclocal/ax_cxx_compile_stdcxx_11.m4'
aclocal: installing 'm4/ax_file_escapes.m4' from '/usr/share/aclocal/ax_file_escapes.m4'
aclocal: installing 'm4/ax_require_defined.m4' from '/usr/share/aclocal/ax_require_defined.m4'
aclocal: error: too many loops
aclocal: Please contact <bug-automake@gnu.org>.
 at /usr/share/automake-1.16/Automake/Channels.pm line 655.
	Automake::Channels::msg("automake", "", "too many loops") called at /usr/share/automake-1.16/Automake/ChannelDefs.pm line 226
	Automake::ChannelDefs::prog_error("too many loops") called at /usr/bin/aclocal line 1187
autoreconf: aclocal failed with exit status: 255
dh_autoreconf: error: autoreconf -f -i returned exit code 1
make: *** [debian/rules:6: build] Error 255
dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2
E: Build command 'cd c-ares-1.17.1 && dpkg-buildpackage -b -uc' failed.
user@debian-build:/veil_rebuild/test$
```

Environment:
OS - Debian 11
Version software:
```
user@debian-build:/veil_rebuild/test$ autoconf --version | head -n1
autoconf (GNU Autoconf) 2.69
user@debian-build:/veil_rebuild/test$ automake --version | head -n1
automake (GNU automake) 1.16.3
user@debian-build:/veil_rebuild/test$ m4 --version | head -n1
m4 (GNU M4) 1.4.18
user@debian-build:/veil_rebuild/test$ libtool --version | head -n1
libtool (GNU libtool) 2.4.6
user@debian-build:/veil_rebuild/test$ cc --version | head -n1
cc (Debian 10.2.1-6) 10.2.1 20210110
user@debian-build:/veil_rebuild/test$ make --version | head -n1
GNU Make 4.3
user@debian-build:/veil_rebuild/test$
```

Output `autoreconf -fi`:
No errors

Output `./configure`:
```
configure: creating ./config.status
config.status: creating Makefile
config.status: creating include/Makefile
config.status: creating src/Makefile
config.status: creating src/lib/Makefile
config.status: creating src/tools/Makefile
config.status: creating docs/Makefile
config.status: creating libcares.pc
config.status: creating src/lib/ares_config.h
config.status: src/lib/ares_config.h is unchanged
config.status: creating include/ares_build.h
config.status: executing depfiles commands
config.status: error: in `/veil_rebuild/test/c-ares-1.17.1':
config.status: error: Something went wrong bootstrapping makefile fragments
    for automatic dependency tracking.  If GNU make was not used, consider
    re-running the configure script with MAKE="gmake" (or whatever is
    necessary).  You can also try re-running configure with the
    '--disable-dependency-tracking' option to at least be able to build
    the package (albeit without support for automatic dependency tracking).
See `config.log' for more details
```

**Comment by dpvpro** [comment]
When i try build c-ares from git i got output:

```
user@debian-build:/veil_rebuild/test$ git clone https://github.com/c-ares/c-ares.git
Cloning into 'c-ares'...
remote: Enumerating objects: 9844, done.
remote: Counting objects: 100% (86/86), done.
remote: Compressing objects: 100% (72/72), done.
remote: Total 9844 (delta 36), reused 29 (delta 14), pack-reused 9758
Receiving objects: 100% (9844/9844), 5.79 MiB | 7.46 MiB/s, done.
Resolving deltas: 100% (5753/5753), done.

user@debian-build:/veil_rebuild/test$ cd c-ares

user@debian-build:/veil_rebuild/test/c-ares$ autoreconf -fi
aclocal: installing 'm4/ax_ac_append_to_file.m4' from '/usr/share/aclocal/ax_ac_append_to_file.m4'
aclocal: installing 'm4/ax_ac_print_to_file.m4' from '/usr/share/aclocal/ax_ac_print_to_file.m4'
aclocal: installing 'm4/ax_add_am_macro_static.m4' from '/usr/share/aclocal/ax_add_am_macro_static.m4'
aclocal: installing 'm4/ax_am_macros_static.m4' from '/usr/share/aclocal/ax_am_macros_static.m4'
aclocal: installing 'm4/ax_check_gnu_make.m4' from '/usr/share/aclocal/ax_check_gnu_make.m4'
aclocal: overwriting 'm4/ax_code_coverage.m4' with '/usr/share/aclocal/ax_code_coverage.m4'
aclocal: installing 'm4/ax_cxx_compile_stdcxx.m4' from '/usr/share/aclocal/ax_cxx_compile_stdcxx.m4'
aclocal: overwriting 'm4/ax_cxx_compile_stdcxx_11.m4' with '/usr/share/aclocal/ax_cxx_compile_stdcxx_11.m4'
aclocal: installing 'm4/ax_file_escapes.m4' from '/usr/share/aclocal/ax_file_escapes.m4'
aclocal: installing 'm4/libtool.m4' from '/usr/share/aclocal/libtool.m4'
aclocal: installing 'm4/ltoptions.m4' from '/usr/share/aclocal/ltoptions.m4'
aclocal: installing 'm4/ltsugar.m4' from '/usr/share/aclocal/ltsugar.m4'
aclocal: installing 'm4/ltversion.m4' from '/usr/share/aclocal/ltversion.m4'
aclocal: installing 'm4/lt~obsolete.m4' from '/usr/share/aclocal/lt~obsolete.m4'
aclocal: installing 'm4/ax_require_defined.m4' from '/usr/share/aclocal/ax_require_defined.m4'
aclocal: error: too many loops
aclocal: Please contact <bug-automake@gnu.org>.
 at /usr/share/automake-1.16/Automake/Channels.pm line 655.
	Automake::Channels::msg("automake", "", "too many loops") called at /usr/share/automake-1.16/Automake/ChannelDefs.pm line 226
	Automake::ChannelDefs::prog_error("too many loops") called at /usr/bin/aclocal line 1187
autoreconf: aclocal failed with exit status: 255
user@debian-build:/veil_rebuild/test/c-ares$
```

**Comment by bradh352** [comment]
Doing a version comparison combined with the error output, it looks like an internal error encountered by automake 1.16.3.  automake 1.16.1 included with CentOS 8 doesn't encounter this issue.

Obviously this version of automake works with other projects, so it must be an edge case triggered by the way c-ares does something, but I'm not entirely sure how to begin trying to work around an internal issue in automake.

The c-ares autotools files really haven't been modified much in the last 5+ years, so I don't think there is any indication this is a regression within c-ares.

In short, I'm not sure where to begin other than actually having someone reach out to the automake team.

**Comment by bradh352** [comment]
I should also mention I strongly encourage everyone to use our CMake build system :)

**Comment by dpvpro** [comment]
Thanks for the quick response
I create bug for automake - https://debbugs.gnu.org/cgi/bugreport.cgi?bug=49901

**Comment by felixonmars** [comment]
I seem to get the same `aclocal: error: too many loops` errors on Arch Linux x86_64:
```
$ autoreconf -ifv
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force -I m4 --install
aclocal: installing 'm4/ax_ac_append_to_file.m4' from '/usr/share/aclocal/ax_ac_append_to_file.m4'
aclocal: installing 'm4/ax_ac_print_to_file.m4' from '/usr/share/aclocal/ax_ac_print_to_file.m4'
aclocal: installing 'm4/ax_add_am_macro_static.m4' from '/usr/share/aclocal/ax_add_am_macro_static.m4'
aclocal: installing 'm4/ax_am_macros_static.m4' from '/usr/share/aclocal/ax_am_macros_static.m4'
aclocal: installing 'm4/ax_check_gnu_make.m4' from '/usr/share/aclocal/ax_check_gnu_make.m4'
aclocal: overwriting 'm4/ax_code_coverage.m4' with '/usr/share/aclocal/ax_code_coverage.m4'
aclocal: installing 'm4/ax_cxx_compile_stdcxx.m4' from '/usr/share/aclocal/ax_cxx_compile_stdcxx.m4'
aclocal: overwriting 'm4/ax_cxx_compile_stdcxx_11.m4' with '/usr/share/aclocal/ax_cxx_compile_stdcxx_11.m4'
aclocal: installing 'm4/ax_file_escapes.m4' from '/usr/share/aclocal/ax_file_escapes.m4'
aclocal: installing 'm4/libtool.m4' from '/usr/share/aclocal/libtool.m4'
aclocal: installing 'm4/ltoptions.m4' from '/usr/share/aclocal/ltoptions.m4'
aclocal: installing 'm4/ltsugar.m4' from '/usr/share/aclocal/ltsugar.m4'
aclocal: installing 'm4/ltversion.m4' from '/usr/share/aclocal/ltversion.m4'
aclocal: installing 'm4/lt~obsolete.m4' from '/usr/share/aclocal/lt~obsolete.m4'
aclocal: installing 'm4/ax_require_defined.m4' from '/usr/share/aclocal/ax_require_defined.m4'
aclocal: error: too many loops
aclocal: Please contact <bug-automake@gnu.org>.
 at /usr/share/automake-1.16/Automake/Channels.pm line 655.
        Automake::Channels::msg("automake", "", "too many loops") called at /usr/share/automake-1.16/Automake/ChannelDefs.pm line 226
        Automake::ChannelDefs::prog_error("too many loops") called at /usr/bin/aclocal line 1187
autoreconf: error: aclocal failed with exit status: 255
```
Versions of the tools:
autoconf 2.71
automake 1.16.4
m4 1.4.19

**Comment by bradh352** [comment]
@felixonmars I see in #417 you must have gotten past this issue, what did you do?

**Comment by felixonmars** [comment]
> @felixonmars I see in #417 you must have gotten past this issue, what did you do?

It went past this `error: too many loops` error on a second run. However, at the end of `configure` script I still get the same `error: Something went wrong bootstrapping makefile fragments` as OP.

It wasn't a usable workaround but I think it's enough for checking the content of generated `ares_build.h` file.

**Comment by bradh352** [comment]
Ah, ok, yeah.  build doesnt actually work, which is what we see after second run.

**Comment by dpvpro** [comment]
With latest code from git i have issue below:
```
user@debian-build:/veil_rebuild/test/c-ares$ autoreconf -fi
aclocal: overwriting '../m4/ax_pthread.m4' with '/usr/share/aclocal/ax_pthread.m4'
libtoolize: putting auxiliary files in '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, '../m4'.
libtoolize: copying file '../m4/libtool.m4'
libtoolize: copying file '../m4/ltoptions.m4'
libtoolize: copying file '../m4/ltsugar.m4'
libtoolize: copying file '../m4/ltversion.m4'
libtoolize: copying file '../m4/lt~obsolete.m4'
configure.ac:12: installing './compile'
configure.ac:12: installing './config.guess'
configure.ac:12: installing './config.sub'
configure.ac:6: installing './install-sh'
configure.ac:6: installing './missing'
Makefile.am: installing './depcomp'
parallel-tests: installing './test-driver'
libtoolize: putting auxiliary files in '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
libtoolize: Remember to add 'LT_INIT' to configure.ac.
configure.ac:495: error: possibly undefined macro: AC_CHECK_INCLUDES_DEFAULT
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
autoreconf: /usr/bin/autoconf failed with exit status: 1
user@debian-build:/veil_rebuild/test/c-ares$
```

**Comment by bradh352** [comment]
please try the latest main branch.  I've tested on quite a few systems and it seems to be working now.

**Comment by dpvpro** [comment]
yes, it's true
